Mathlib/RingTheory/Algebraic/Basic.lean

```diff
@@ -17,39 +15,20 @@ The main result in this file proves transitivity of algebraicity:
 a tower of algebraic field extensions is algebraic.
 -/
 
-
 universe u v w
 
-open scoped Classical
 open Polynomial
 
 section
 
 variable (R : Type u) {A : Type v} [CommRing R] [Ring A] [Algebra R A]
 
-/-- An element of an R-algebra is algebraic over R if it is a root of a nonzero polynomial
-with coefficients in R. -/
-@[stacks 09GC "Algebraic elements"]
-def IsAlgebraic (x : A) : Prop :=
-  ∃ p : R[X], p ≠ 0 ∧ aeval x p = 0
-
-/-- An element of an R-algebra is transcendental over R if it is not algebraic over R. -/
-def Transcendental (x : A) : Prop :=
-  ¬IsAlgebraic R x
-
 @[nontriviality]
 theorem is_transcendental_of_subsingleton [Subsingleton R] (x : A) : Transcendental R x :=
   fun ⟨p, h, _⟩ => h <| Subsingleton.elim p 0
```

Comment:
Do you think it's a good idea to move this result as well as some other simple results depends only on `Mathlib.Algebra.Polynomial.AlgebraMap` to `Defs.lean`?

Reply:
This is always a bit of a tradeoff. One value of keeping `Defs.lean` small is for minimizing: especially if you replace all proofs with `sorry` then reducing the imports to only `Defs` makes things a lot easier. On the other hand, we would like simple results to be easily accessible.

So I think it's generally a good idea to be quite strict with `Defs.lean` and try to avoid all theory that are not directly needed for other definitions.

Reply:
OK that's fine. In fact I have two new results for `Transcendental` added in #19527. I've checked that they could also be added to `Defs.lean`, but it's up to you to decide where should they go.

Reply:
I did notice them, neat proofs! I think again we should try to avoid adding too much theory to Defs files so I put them in `Basic.lean`.